src/main/java/frc/robot/utils/InterpolateDouble.java

```diff
@@ -0,0 +1,34 @@
+package frc.robot.utils;
+
+import java.util.HashMap;
+
+public class InterpolateDouble {
+    private HashMap<Double, Double> map;
+
+    public InterpolateDouble(HashMap<Double, Double> map) {
+        this.map = map;
+    }
+
+    public double getValue(double key) {
+        if (map.containsKey(key)) {
+            return map.get(key);
+        }
+
+        double lowerKey = 0;
+        double upperKey = 0;
+        for (double k : map.keySet()) {
```

Comment:
Does keySet return the keys in order? If yes, feel free to resolve this comment

Reply:
It does not but I put a note in the constant. I also now sorted the keys just in case.